installers/npm/proxy.js

```diff
@@ -0,0 +1,92 @@
+const formatHostName = (hostname) =>
```

Comment:
hmmm... i wonder - can we just add `request` as a dependency and use that directly? this code is a bit gnarly

Reply:
request module doesn't export that method/utility.
Also, request module is deprecated/not supported anymore.

Other way could be : move away from axios implementation in binary-install module or use some other module.


I think as it's just a simple utility which should be fine

Reply:
@EverlastingBugstopper I don't know if it will end up being any more maintainable, but I moved the proxy functionality into an npm package and typescript'ified it and added some tests. Is this worth patching in? https://github.com/farawaysouthwest/axios-proxy-builder

Reply:
wow! that's awesome! yeah i say let's go for it. we may in the future want to move to something like `fetch` but this is fantastic for now.